sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -72,6 +72,12 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   def requiredChildDistribution: Seq[Distribution] =
     Seq.fill(children.size)(UnspecifiedDistribution)
 
+  /** Specifies how data is ordered in each partition. */
+  def outputOrdering: Option[Ordering[Row]] = None
```

Comment:
Why did you change this from `Seq[SortOrder]`.  I think that was much better as that exposes the logical expressions that define the order.  `Ordering` is an opaque scala class that may or may not implement a reasonable equality function.